sdk/core/core-rest-pipeline/documentation/pipeline.md

```diff
@@ -0,0 +1,220 @@
+# Policies
+
+A pipeline policy manipulates a request as it travels through the pipeline. It is conceptually a middleware that is allowed to modify the request before it is made as well as the response when it is received.
+
+```typescript
+export type SendRequest = (request: PipelineRequest) => Promise<PipelineResponse>;
+
+export interface PipelinePolicy {
+  name: string;
+  sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse>;
+}
+```
+
+It is given a string name so that consumers can express dependencies relative to existing policies. The `sendRequest` method uses a middleware pattern, which is familiar to JS developers.
```

Comment:
referencing the middleware pattern here feels slightly redundant to the sentence above the block, but maybe we should also link out to some description of how middleware works?